code/game/atoms.dm

```diff
@@ -1415,6 +1415,43 @@
 	if(source != target)
 		target.log_talk(message, message_type, tag="[tag] from [key_name(source)]", log_globally=FALSE)
 
+/**
+  * Log for crafting items
+  *
+  * 1 argument is for the user making the item
+  * 2 argument is for the item being created
+  * 3 argument is whether or not the user has finished creating it
+ */
+/proc/log_crafting(atom/user, atom/object, finished=FALSE)
+	var/masterpiece = object
+	var/message = "has [finished ? "finished":"started"] crafting [masterpiece]"
+	user.log_message(message, LOG_GAME)
+	var/dangerous_crafts = typecacheof(list( //List of items that make an announcement to the admins when crafted
```

Comment:
make this a global or static so it's not initialized every time you call the proc.

Reply:
Done